tests/extension/oneapi_root_group/root_group.cpp

```diff
@@ -0,0 +1,256 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2023 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "../../common/common.h"
+
+namespace root_group::tests {
+
+#ifdef SYCL_EXT_ONEAPI_ROOT_GROUP
+
+template <typename KernelName, size_t... Dims>
+static void check_root_group_api() {
+  constexpr int Dimensions = sizeof...(Dims);
+  auto q = sycl_cts::util::get_cts_object::queue();
+  auto bundle =
+      sycl::get_kernel_bundle<sycl::bundle_state::executable>(q.get_context());
+  auto kernel = bundle.get_kernel<KernelName>();
+  auto maxWGs = kernel.template ext_oneapi_get_info<
+      sycl::ext::oneapi::experimental::info::kernel_queue_specific::
+          max_num_work_group_sync>(q);
+  REQUIRE(maxWGs >= 1);
```

Comment:
I don't think that the query or the `use_root_sync`  property is necessary for this test, according to the [extension spec](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_root_group.asciidoc#overview):

> An instance of the `root_group` class can be accessed by any ND-range kernel, since all information it exposes via its member functions is equivalent to information already available via members of `sycl::nd_item`.

Therefore, I would that that it makes sense to check that `root_group` class is still accessible and returns the right values regardless of the property or launch configuration. Perhaps, this particular test case can have a few variations like with, or without the property.

Tagging @Pennycook here for additional opinion on the matter.

Reply:
I agree that we should be checking the API works for kernels without the root-sync property.

If we want to cut down on the number of combinations to check, we could run the API test at the same time as the barrier test.  Then we'd only need two tests: one that checks the API works in kernels without the root-sync property; and another that checks the API (including barriers) works in kernels with the root-sync property.